internal/services/apimanagement/api_management_custom_domain_resource_test.go

```diff
@@ -93,6 +94,46 @@ func TestAccApiManagementCustomDomain_update(t *testing.T) {
 	})
 }
 
+func TestAccApiManagementCustomDomain_certificateManaged(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_custom_domain", "test")
+	r := ApiManagementCustomDomainResource{}
+
+	if os.Getenv("ARM_TEST_DNS_ZONE") == "" {
+		t.Skipf("Skipping managed custom domain test as ARM_TEST_DNS_ZONE is not set.")
+	}
+
+	if os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP") == "" {
+		t.Skipf("Skipping managed custom domain test as ARM_TEST_DNS_ZONE_RESOURCE_GROUP is not set.")
+	}
+
+	if os.Getenv("ARM_TEST_DNS_ZONE_TXT_RECORD") == "" {
+		t.Skipf("Skipping managed custom domain test as ARM_TEST_DNS_ZONE_TXT_RECORD is not set.")
+	}
```

Comment:
This can all be wrapped in a `preCheck` function which each test could call instead of the multiple inline hardcoded environment `os` checks...